### PR TITLE
new component: pcal9535a

### PR DESCRIPTION
**Pull request by Shulyaka**
Signed-off-by: Denis Shulyaka <Shulyaka@gmail.com>

**Description:** Support for PCAL9535A I2C GPIO PIN expander

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26563

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/pcal9535a.markdown [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+logo: raspberry-pi.png
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: 0.101
+ha_iot_class: Local Polling
+---
+
+The `pcal9535a` integration is the base for all related pcal9535a platforms in Home Assistant. There is no setup needed for the integration itself, for the platforms please check their corresponding sections.
```
Comment (frenck) [inline source/_components/pcal9535a.markdown]:
```suggestion
The `pcal9535a` integration is the base for all related pcal9535a platforms in Home Assistant. There is no setup needed for the integration itself, for the platforms, please check their corresponding sections.
```

File: source/_components/pcal9535a.markdown [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+One of the use cases is [Seeed studio Raspberry Pi Relay Board](http://wiki.seeedstudio.com/Raspberry_Pi_Relay_Board_v1.0/).
+
+For more details about the PCAL9535A I2C I/O port expander you can find its datasheet here: [PCAL9535A](https://www.nxp.com/docs/en/data-sheet/PCAL9535A.pdf).
```
Comment (frenck) [inline source/_components/pcal9535a.markdown]:
```suggestion
For more details about the PCAL9535A I2C I/O port expander, you can find its datasheet here: [PCAL9535A](https://www.nxp.com/docs/en/data-sheet/PCAL9535A.pdf).
```

File: source/_components/pcal9535a.markdown [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+## Binary Sensor
+
+The `pcal9535a` binary sensor platform allows you to read sensor values from the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```
Comment (frenck) [inline source/_components/pcal9535a.markdown]:
```suggestion
The pin numbers are from 0 to 15, where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```

File: source/_components/pcal9535a.markdown [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  required: false
+  type: string
+  default: "`DISABLED`"
+{% endconfiguration %}
+
+## Switch
+
+The `pcal9535a` switch platform allows you to write to the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```
Comment (frenck) [inline source/_components/pcal9535a.markdown]:
```suggestion
The pin numbers are from 0 to 15, where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```

File: source/_components/pcal9535a.markdown [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### Configuration
+
+To use the I/O pins of an pcal9535a connected to an I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+  - platform: pcal9535a
+    i2c_address: 0x20
```
Comment (frenck) [inline source/_components/pcal9535a.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

This applies to all other optional fields in this PR as well.

**Review by frenck — APPROVED** [review]
Thanks for the PR @Shulyaka! 

I've reviewed it and left some comments. Could you please take a look? Thanks! 👍

**Review by frenck — CHANGES_REQUESTED** [review]
Hit the wrong button, let me try again.

File: source/_components/pcal9535a.markdown (hunk 3 above)
Comment (Shulyaka) [reply source/_components/pcal9535a.markdown]: Are you sure it should be singular? We are talking about several pins (0 to 7) which correspond to the port.

File: source/_components/pcal9535a.markdown (hunk 5 above)
Comment (Shulyaka) [inline source/_components/pcal9535a.markdown]: Ok, I understand that point, I just thought that this parameter was important to verify even though it has a default value.
```suggestion
```

File: source/_components/pcal9535a.markdown [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+### Configuration
+
+To use the I/O pins of an pcal9535a connected to an I2C bus of your Raspberry Pi as switches, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+switch:
+  - platform: pcal9535a
+    i2c_address: 0x20
```
Comment (Shulyaka) [inline source/_components/pcal9535a.markdown]:
```suggestion
```

File: source/_components/pcal9535a.markdown (hunk 3 above)
Comment (frenck) [reply source/_components/pcal9535a.markdown]: Sorry, you are right, adjusted suggestion.

File: source/_components/pcal9535a.markdown (hunk 4 above)
Comment (frenck) [reply source/_components/pcal9535a.markdown]: Adjusted suggestion.

File: source/_components/pcal9535a.markdown (hunk 5 above)
Comment (frenck) [reply source/_components/pcal9535a.markdown]: You could add an extra note, or provide a full example as an alternative.

File: source/_components/pcal9535a.markdown (hunk 6 above)
Comment (frenck) [reply source/_components/pcal9535a.markdown]: You have to apply suggestions or make adjustments yourself.
It is your fork in your GitHub account 😉

File: source/_components/pcal9535a.markdown (hunk 3 above)
Comment (Shulyaka) [reply source/_components/pcal9535a.markdown]: I'm changing it for now as requested, but I still believe it might be wrong:) However English is not my native language.

File: source/_components/pcal9535a.markdown [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+One of the use cases is [Seeed studio Raspberry Pi Relay Board](http://wiki.seeedstudio.com/Raspberry_Pi_Relay_Board_v1.0/).
+
+For more details about the PCAL9535A I2C I/O port expander, you can find its datasheet here: [PCAL9535A](https://www.nxp.com/docs/en/data-sheet/PCAL9535A.pdf).
+
+## Binary Sensor
+
+The `pcal9535a` binary sensor platform allows you to read sensor values from the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15, where: 0-7 corresponds to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```
Comment (Shulyaka) [inline source/_components/pcal9535a.markdown]:
```suggestion
The pin numbers are from 0 to 15, where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```

File: source/_components/pcal9535a.markdown [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  required: false
+  type: string
+  default: "`DISABLED`"
+{% endconfiguration %}
+
+## Switch
+
+The `pcal9535a` switch platform allows you to write to the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15, where: 0-7 corresponds to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```
Comment (Shulyaka) [inline source/_components/pcal9535a.markdown]:
```suggestion
The pin numbers are from 0 to 15, where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
```

File: source/_components/pcal9535a.markdown (hunk 3 above)
Comment (Shulyaka) [reply source/_components/pcal9535a.markdown]: Fixed

File: source/_components/pcal9535a.markdown (hunk 6 above)
Comment (Shulyaka) [reply source/_components/pcal9535a.markdown]: I made this 'suggestion' to myself just in order to apply all changes in one batch in a single commit.

**Review by frenck — APPROVED** [review]
Thanks, @Shulyaka! 👍 

I've made a couple of small tweaks, furthermore, I've removed the `scan_interval` option, since it is not part of this integration, but part of the platform it uses.

LGTM now! 👍
